### ISSUE TITLE
SSH connections should validate host and work with user's system settings

### ISSUE THREAD

**Issue by mjtravers** [ReproNim/niceman#177]
In resources/ssh.py we shouldn't blindly login to unknown hosts.

Also:
didn't look how difficult or possible via paramiko but we need make sure that it works coherently with existing system/user ssh configuration, i.e. that existing known_hosts used by user's ssh command are used, which seems to be not the case in simplest paramiko invocation:

```
$> python -c 'from paramiko import SSHClient; ssh = SSHClient().connect("127.0.0.1", username=None)'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python2.7/dist-packages/paramiko/client.py", line 361, in connect
    server_key)
  File "/usr/lib/python2.7/dist-packages/paramiko/client.py", line 648, in missing_host_key
    raise SSHException('Server %r not found in known_hosts' % hostname)
paramiko.ssh_exception.SSHException: Server '127.0.0.1' not found in known_hosts
```

* make sure that ssh keys authentication (possibly provided by ssh-agent) would be used

**Comment by yarikoptic** [comment]
@memartone Please have a look at https://github.com/paramiko/paramiko/blob/master/demos/demo.py#L41 which seems to provide quite a complete example for establishing an interactive session, while also trying to use SSHAgent if available.  It worked for me, besides only not picking up custom `port` setting for the host in my `.ssh/config` (yet to check if that is complexity/mess of my config or something else)

**Comment by mjtravers** [comment]
Addressed in PR #264
